src/homeblks/volume/volume.cpp

```diff
@@ -335,7 +335,96 @@ indx_tbl* Volume::recover_indx_tbl(btree_super_block& sb, btree_cp_sb& cp_info)
     return static_cast< indx_tbl* >(tbl);
 }
 
+static bool is_buf_zero(const uint8_t* buf, size_t size) {
+    // TODO: subsample the buffer to detect zero request instead of working on the whole buffer to achieve constant
+    //  processing time for large buffer size requests. Needs to investigate the performance impact of this change
+    //  in end2end testing.
+    static std::vector< uint8_t > read_buf(size, 0);
+    static const auto zero_crc = crc16_t10dif(init_crc_16, read_buf.data(), size);
+    const auto crc = crc16_t10dif(init_crc_16, buf, size);
+    return (crc == zero_crc) ? (buf[0] == 0 && !std::memcmp(buf, buf + 1, size - 1)) : false;
+}
+
+static bool is_iovec_zero(const std::vector< iovec >& iovecs) {
+    for (const auto& iovec : iovecs) {
+        auto data = static_cast< uint8_t* >(iovec.iov_base);
+        const size_t size = iovec.iov_len;
+        if (!is_buf_zero(data, size)) { return false; }
+    }
+    return true;
+}
+
+static bool is_zero_request(const vol_interface_req_ptr& iface_req, uint32_t page_size) {
+    if (iface_req->iovecs.empty()) {
+        return is_buf_zero(static_cast< uint8_t* >(iface_req->buffer), iface_req->nlbas * page_size);
```

Comment:
Is there already a place checking iface_req->buffer won't be nullptr when iovecs is empty?